sqlglot/generator.py

```diff
@@ -2888,7 +2888,12 @@ def reference_sql(self, expression: exp.Reference) -> str:
         return f"REFERENCES {this}{expressions}{options}"
 
     def anonymous_sql(self, expression: exp.Anonymous) -> str:
-        return self.func(self.sql(expression, "this"), *expression.expressions)
+        # We don't normalize qualified functions such as a.b.foo(), because they can be case-sensitive
+        parent = expression.parent
+        is_qualified = isinstance(parent, exp.Dot) and expression is parent.expression
```

Comment:
Could this be simplified by checking `arg_key`? 

Also, is this a problem for other dialects too e.g. Oracle's `getxmltype`?

Reply:
> Could this be simplified by checking arg_key?

I don't think so, it's essentially the same check and involves a string comparison vs object equivalence.

> Also, is this a problem for other dialects too e.g. Oracle's getxmltype?

Dunno but the most conservative approach is to assume it is, since you'll just get identity for roundtrip.